### PR TITLE
cargo-spellcheck: init at 0.8.13

### PR DESCRIPTION
**Pull request by newAM**
###### Motivation for this change

Add [cargo-spellcheck](https://github.com/drahnr/cargo-spellcheck), a tool to check rust documentation for spelling and grammar mistakes.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/rust/cargo-spellcheck/default.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    export LIBCLANG_PATH="${libclang.lib}/lib"
+  '';
+
+  checkFlags = [
+    "--skip checker::hunspell::tests::hunspell_binding_is_sane"
+  ];
+
+  nativeBuildInputs = [ pkg-config rustfmt ];
+  buildInputs = [ hunspell libclang ];
+
```
Comment (figsoda) [inline pkgs/development/tools/rust/cargo-spellcheck/default.nix:33]: hunspell comes bundled with the cargo dependency
```suggestion
```

File: pkgs/development/tools/rust/cargo-spellcheck/default.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+, hunspell
+, pkg-config
+, rustfmt
```
Comment (figsoda) [inline pkgs/development/tools/rust/cargo-spellcheck/default.nix:6]:
```suggestion
```

**Review by figsoda — CHANGES_REQUESTED** [review]

**Comment by newAM** [comment]
I made the suggested changes, thank you for the review!

**Comment by figsoda** [comment]
@ofborg eval

**Comment by figsoda** [comment]
Result of `nixpkgs-review pr 138458` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>cargo-spellcheck</li>
  </ul>
</details>

**Review by figsoda — APPROVED** [review]

**Pull request merged**